File: rocket-bi-server/src/main/scala/co/datainsider/jobworker/domain/source/HubspotSource.scala
package co.datainsider.jobworker.domain.source

import co.datainsider.jobworker.domain.{DataSource, DataSourceType}
import co.datainsider.jobworker.domain.DataSourceType.DataSourceType
import co.datainsider.jobworker.domain.Ids.SourceId

case class HubspotSource(
    id: SourceId,
    displayName: String,
    apiKey: String
) extends DataSource {
  override def getId: SourceId = id

  override def getName: String = displayName

  override def getType: DataSourceType = DataSourceType.Hubspot

  override def getConfig: Map[String, Any] = Map()
}

File: web/di-core/data-ingestion/domain/response/HubspotSource.ts
import { DataSources } from '@core/data-ingestion';
import { DataSource } from '@core/data-ingestion/domain/response/DataSource';
import { SourceId } from '@core/common/domain';

export class HubspotSource implements DataSource {
  readonly className = DataSources.Hubspot;
  id: SourceId;
  orgId: string;
  displayName: string;
  apiKey: string;
  lastModify: number;

  constructor(id: SourceId, orgId: string, displayName: string, apiKey: string, lastModify: number) {
    this.id = id;
    this.orgId = orgId;
    this.displayName = displayName;
    this.apiKey = apiKey;
    this.lastModify = lastModify;
  }

  static fromObject(obj: any): HubspotSource {
    return new HubspotSource(obj.id, obj.orgId, obj.displayName, obj.apiKey, obj.lastModify);
  }
}
